Add controllers

Kazantsevyury opened the pull request:
Initialized app framework and web layer focusing on Item entity, enabling users to add, describe, and set rental status for items. Introduced search functionality, booking entity for reservations, and feedback mechanism post-rental. Created ItemRequest for unlisted items, organized code into feature-based packages (item, booking, request, user), and added DTOs and Mappers f

Kazantsevyury committed:
feat: Add app skeleton and web layer

Initialized app framework and web layer focusing on Item entity, enabling users to add, describe, and set rental status for items. Introduced search functionality, booking entity for reservations, and feedback mechanism post-rental. Created ItemRequest for unlisted items, organized code into feature-based packages (item, booking, request, user), and added DTOs and Mappers for entity handling. Set the stage for database integration in future sprints.

Kazantsevyury committed:
checkstyle

Kazantsevyury committed:
checkstyle

Kazantsevyury committed:
checkstyle

Kazantsevyury committed:
checkstyle

Kazantsevyury committed:
checkstyle

Kazantsevyury committed:
checkstyle

Kazantsevyury committed:
checkstyle

Kazantsevyury committed:
checkstyle

Kazantsevyury committed:
checkstyle

Kazantsevyury committed:
checkstyle

Kazantsevyury committed:
checkstyle

Kazantsevyury committed:
checkstyle

Kazantsevyury committed:
checkstyle

Kazantsevyury committed:
checkstyle

Kazantsevyury committed:
checkstyle

Kazantsevyury committed:
checkstyle

Kazantsevyury committed:
checkstyle

Kazantsevyury committed:
checkstyle

Kazantsevyury committed:
checkstyle

PolnySkvorets commented on src/main/java/ru/practicum/shareit/exception/GlobalExceptionHandler.java:
```diff
@@ -0,0 +1,19 @@
+import ru.practicum.shareit.exception.exceptions.AccessDeniedException;
+
+@ControllerAdvice
+public class GlobalExceptionHandler {
```
Здесь можно также отлавливать другие исключения, например entityNotFound, и не описывать эту логику в контроллере

PolnySkvorets commented on src/main/java/ru/practicum/shareit/item/ItemController.java:
```diff
@@ -1,12 +1,120 @@
+            return ResponseEntity.badRequest().body(errors.getAllErrors());
+        }
+
+        if (itemDto.getAvailable() == null || Boolean.FALSE.equals(itemDto.getAvailable())) {
```
Это лучше вынести в валидатор отдельным методом

PolnySkvorets reviewed (commented):
Привет! Посмотри замечания. Можно заиспользовать механизм обработки исключений и не описывать большую логику валидации в контроллере

Kazantsevyury committed:
refactor: streamline ItemController and UserController

- Move validation logic from controllers to Item and User models using ItemValidator and UserValidator.
- Relocate error handling to UserServiceImpl and ItemServiceImpl services for cleaner controller operations.
- Enhance and integrate ExceptionFactory and GlobalExceptionHandler more tightly into the application.
- Implement LoggingAspect for improved logging and extend logging coverage for complex parts of the program.

Kazantsevyury committed:
Merge remote-tracking branch 'origin/add-controllers' into add-controllers

PolnySkvorets reviewed (approved):
Привет! Добавление Аспектов - это конечно круто. Принято
